libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -521,7 +522,7 @@ void AP_Baro::update(void)
             // update altitude calculation
             float ground_pressure = sensors[i].ground_pressure;
             if (!is_positive(ground_pressure) || isnan(ground_pressure) || isinf(ground_pressure)) {
-                sensors[i].ground_pressure = sensors[i].pressure;
```

Comment:
The reason this doesn't work on plane is even if you only hit this because of a sensor error (lets suppose it's momentary), this would create a baro pressure reading that throws the aircraft up possibly thousands of feet. Once the baro altitude is greater then 2 meters plane will stop automatically calibrating the baro, which means that outisde of a user starting an airspeed cal they are now stuck with an insane altitude, which they hopefully noticed before taking off.

Ground pressure is just the relative starting point for all altitude caluclations, you simply can't mess with it in flight ever without creating a step input, and all attempts to do so (short of params) should be rejected (I think they are) while the aircraft is armed. Ground pressure should always be jumped to the current pressure when doing a calibration at least for everything other then sub.